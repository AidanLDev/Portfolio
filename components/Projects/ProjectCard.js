import React, { useState } from 'react'
import styles from './style.module.scss'
import { Image, Tooltip, Text } from '@chakra-ui/react'
import useIsTablet from '../../hooks/useIsTablet'

import { motion } from 'framer-motion'

export default function ProjectCard({ img, link, tooltip, gitHubLink }) {
  const [isShown, setIsShown] = useState(false)
  const isTablet = useIsTablet()
  return (
    <div
      className={styles.projectCard}
      onMouseEnter={() => setIsShown(true)}
      onMouseLeave={() => setIsShown(false)}
    >
      <a href={link} rel="noreferrer" target="_blank">
        {gitHubLink && isShown && !isTablet && (
          <Tooltip label="View code on GitHub">
            <div className={styles.ghOverlay}>
              <a href={gitHubLink} rel="noreferrer" target="_blank">
                <Image
                  w="64px"
                  as={motion.img}
                  alt="GitHub Logo"
                  src={'/images/Logos/GithubLogo.png'}
                  whileHover={{ scale: 1.1 }}
                />
              </a>
            </div>
          </Tooltip>
        )}
        {gitHubLink && isTablet && (
          <div className={styles.ghOverlay}>
            <a href={gitHubLink} rel="noreferrer" target="_blank">
              <Image
                w="64px"
                as={motion.img}
                alt="GitHub Logo"
                src={'/images/Logos/GithubLogo.png'}
                whileHover={{ scale: 1.1 }}
              />
              {/* <Text>
                View code on GitHub
              </Text> */}
            </a>
          </div>
        )}
        <Tooltip label={tooltip}>
          <Image
            as={motion.img}
            alt={tooltip}
            src={`/images/Projects/${img}`}
            h={240}
            width="100%"
            whileHover={{ scale: 1.1 }}
          />
        </Tooltip>
      </a>
    </div>
  )
}
