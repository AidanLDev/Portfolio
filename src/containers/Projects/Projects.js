import React from 'react'
import Card from 'components/Card/Card'
import BurgerBuilderImg from 'assets/Burger.png'
import * as Descriptions from 'data/Data'

import styles from './Style'

const Projects = props => {

  return (
    <div>
       {Descriptions.projectsDescription}
      <div className={styles.projectCards}>
      
        <Card
          title="Burger Builder"
          content={Descriptions.burgerBuilderDesc}
          img={BurgerBuilderImg}
          imgLink
          imgWidth={300}
          imgHeight={200}
          imgLinkLocation="https://aidans-burger-builder.firebaseapp.com/"
          imgAlt="Burger builder project"
        />
        <Card
          title="Burger Builder"
          content={Descriptions.burgerBuilderDesc}
          img={BurgerBuilderImg}
          imgLink
          imgWidth={300}
          imgHeight={200}
          imgLinkLocation="https://aidans-burger-builder.firebaseapp.com/"
          imgAlt="Burger builder project"
        />
      </div>
    </div>
  )
}

export default Projects;