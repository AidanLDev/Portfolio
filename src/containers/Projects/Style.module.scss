@import"styles/pallet.scss";
@import"styles/variables.scss";
@import"styles/mixins.scss";

.projectCards {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  .userAuthCard {
    img {
      box-shadow: 0 2px 1px $darkGrey;
      border: 2px solid $darkGrey;
      border-radius: 10px;
    }
  }
}
