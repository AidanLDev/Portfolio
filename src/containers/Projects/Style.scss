@import "pallet";
@import "variables";
@import "mixins";

.projectCards {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;

  .userAuthCard {
    img {
      box-shadow: 0 2px 1px $darkGrey;
      border: 2px solid $darkGrey;
      border-radius: 10px;
    }
  }
}
