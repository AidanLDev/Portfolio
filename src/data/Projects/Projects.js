import React from 'react'

import styles from './Style'

export const projectsDescription = <p>
Developing websites/web applications is more than a carear, but a hobby too. I have worked on various personal projects:
</p>


export const burgerBuilder = <div className={styles.burgerBuilder}>
  <p>The Burger Builder was developed as part of a Udemy course on React 16; including hooks, the context API, Redux and Firebase's NoSQL database. The project highlights user authentication and storing persistant data using Firebase, a FE application using React with state management from Redux. Once authenticated you can create a burger using the controls at the bottom of the page and 'order' this burger. Orderes are stored on a NoSQL database and can be viewed, if authenticated, in the Orders page</p>
  {/* <div className={styles.imgWrapper}></div>
  <img src={firebaseLogo} alt="Firebase logo" width="100" height="100"/>
  <img src={reactLogo} alt="React logo" width="100" height="100"/>
  <img src={reduxLogo} alt="Redux logo" width="130" height="100"/> */}
  </div>

export const nodeShop = <div className={styles.nodeShop}>
  <p>This project uses Node.js, GraphQL, and SQL to build a shop and several templating engines.</p>
  {/* <img src={graphQLLogo} alt="graphQL logo" width="100" height="100"/>
  <img src={nodeLogo} alt="node.js logo" width="100" height="100"/>
  <img src={sqlLogo} alt="sql logo" width="100" height="100"/> */}
</div>

export const userAuth = <div className={styles.userAuth}>
  <p>A project created by team treehouse, teaching Mongoose, the MongoDB framework and Express, the Node.js framework, to create a user authentication system. The project uses hashing and saliting techniques to encrypt sensative user information in a MongoDB.</p>
</div>