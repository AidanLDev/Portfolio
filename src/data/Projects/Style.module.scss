.padding {
  padding: 0 6px 0 6px;
}
