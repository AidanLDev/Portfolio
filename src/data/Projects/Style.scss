.burgerBuilder {
  padding: 0 6px 0 6px;
}

.nodeShop {

}

.userAuth {
  padding: 0 6px 0 6px;
}