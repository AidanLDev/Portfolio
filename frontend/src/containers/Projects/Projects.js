import React from "react";
import Card from "components/Card/Card";
import BurgerBuilderImg from "assets/Burger.png";
import placeHolderImg from "assets/ComingSoon.png";
import * as descriptions from "data/Projects/Projects";

import styles from "./Style";

const Projects = props => {
  return (
    <div>
      <div className={styles.projectCards}>
        <Card
          title="Burger Builder"
          content={descriptions.burgerBuilder}
          img={BurgerBuilderImg}
          imgWidth={300}
          imgHeight={200}
          imgLinkLocation="https://aidans-burger-builder.firebaseapp.com/"
          imgAlt="Burger builder project"
        />
        <Card
          title="Udemy Serverside Shop"
          content={descriptions.nodeShop}
          img={placeHolderImg}
          imgWidth={500}
          imgHeight={200}
          // imgAlt="Serverside Shop"
        />
        <Card
          title="Treehouse user authentication"
          content={descriptions.userAuth}
          img={placeHolderImg}
          imgWidth={500}
          imgHeight={200}
          // transition
          imgLinkLocation="https://aidans-auth-project.herokuapp.com/"
          imgAlt="User Authentication project - Bookworm"
        />
      </div>
    </div>
  );
};

export default Projects;
