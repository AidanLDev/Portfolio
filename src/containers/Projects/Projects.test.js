import React from 'react';
import Projects from './Projects';
import { shallow } from 'enzyme';

it('Renders Projects Component Correctly', () => {
  const component = shallow(<Projects />);
  expect(component).toMatchSnapshot();
});
