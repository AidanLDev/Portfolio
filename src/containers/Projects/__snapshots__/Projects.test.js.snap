// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Projects Component Correctly 1`] = `
<div>
  <div
    className="projectCards"
  >
    <Card
      content={
        <div
          className="padding"
        >
          <p>
            The Burger Builder was developed as part of a Udemy course on React 16; including hooks, the context API, Redux and Firebase's NoSQL database. The project highlights user authentication and storing persistent data using Firebase, a FE application using React with state management from Redux. Once authenticated you can create a burger using the controls at the bottom of the page and 'order' this burger. Orders are stored on a NoSQL database and can be viewed, if authenticated, in the Orders page
          </p>
        </div>
      }
      hoverScale={1.2}
      img="Burger.png"
      imgAlt="Burger builder project"
      imgHeight={200}
      imgLinkLocation="https://aidans-burger-builder.firebaseapp.com/"
      imgWidth={300}
      projects="https://github.com/AidanLDev/Training-Burger-Builder"
      title="Burger Builder"
    />
    <Card
      content={
        <p
          className="padding"
        >
          Using the Mars InSight API that provides data from our red neighbour, I have created a simple webpage that displays what the weathers like on Mars.
        </p>
      }
      img="Mars_space.jpg"
      imgAlt="Burger builder project"
      imgHeight={250}
      imgLinkLocation="https://aidanldev.github.io/"
      imgWidth={400}
      projects="https://github.com/AidanLDev/Weather-App"
      title="Mars Weather App"
    />
    <Card
      content={
        <p
          className="padding"
        >
          Search through a data set of songs, either by searching for a specific track ID or by Artist name. The frontend uses a Bootstrap template and a React Frontend to serve the initiative UI. The app is served using Express and the API for querying and serving the data is also written using Express. Using TypeScript, development for this project was quick as I could utilise VSCode's built-in TypeScript support and reduce bugs that may otherwise have gone unnoticed.
        </p>
      }
      img="MusicLibraryProject.png"
      imgAlt="Burger builder project"
      imgHeight={250}
      imgLinkLocation="https://aidans-song-search.herokuapp.com/"
      imgWidth={400}
      projects="https://github.com/AidanLDev/Song_Search"
      title="Track Search"
    />
    <Card
      content={
        <p
          className="padding"
        >
          Using React16, the frontend framework Bootstrap React and the context API I designed and developed a city guide application. I used a mobile first approach for this project, using the Boostrap framework it's simple to create good looking UI layouts and components. The components are further styled using Styled Components the popular NPM module and React Router.
        </p>
      }
      img="ReactBootstrapCityGuide.png"
      imgAlt="Burger builder project"
      imgHeight={250}
      imgLinkLocation="http://react-bootstrap-city-guide.s3-website.eu-west-2.amazonaws.com/"
      imgWidth={400}
      projects="https://github.com/AidanLDev/React-Bootstrap-City-Guide"
      title="React Bootstrap City Guide"
    />
    <Card
      content={
        <p
          className="padding"
        >
          My take on the classic Fizz Buzz programming challenge. The program takes two numbers It will print Fizz when there's a multiple of the first number Buzz when it's a multiple of the second number. When a multiple of both the first and second number Fizz Buzz will be printed
        </p>
      }
      img="fizzBuzz.png"
      imgAlt="Burger builder project"
      imgHeight={250}
      imgLinkLocation="http://fizz-buzz.s3-website.eu-west-2.amazonaws.com/"
      imgWidth={400}
      projects="https://github.com/AidanLDev/FizzBuzz.js"
      title="FizzBuzz"
    />
  </div>
</div>
`;
